src/hyperion/external_interaction/config_server.py

```diff
@@ -0,0 +1,37 @@
+from typing import TypeVar
+
+from daq_config_server.client import ConfigServer
+from pydantic import BaseModel
+
+from hyperion.log import LOGGER
+from hyperion.parameters.constants import CONST
+
+_CONFIG_SERVER: ConfigServer | None = None
+T = TypeVar("T")
+
+
+def config_server() -> ConfigServer:
+    global _CONFIG_SERVER
+    if _CONFIG_SERVER is None:
+        _CONFIG_SERVER = ConfigServer(CONST.CONFIG_SERVER_URL, LOGGER)
+    return _CONFIG_SERVER
+
+
+def best_effort_get_feature_flag(flag_name: str, fallback: T = None) -> bool | T:
+    return config_server().best_effort_get_feature_flag(flag_name, fallback)
+
+
+class FeatureFlags(BaseModel):
+    # The default value will be used as the fallback when doing a best-effort fetch
+    # from the service
+    use_panda_for_gridscan: bool = False
+    use_gpu_for_gridscan: bool = False
+    set_stub_offsets: bool = False
+
+    @classmethod
+    def best_effort(cls):
```

Comment:
This method gets invoked every time we want only one of the feature flags, but it gets all of the feature flags. Why don't we just get all of the feature flags right at the start and put them all in the parameters in one go?

I realise that while it is a rest call this is just going to likely be quite fast, but at the end of the day we are in the long run ultimately trying to get end-to-end times of < 10s and say a few hundred ms extra if you end up with a lot of calls baked in to the code due to bad API design choices early on, it isn't completely insignificant.